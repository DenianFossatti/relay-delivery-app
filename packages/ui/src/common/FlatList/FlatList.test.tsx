import { mockTestID, render, wrapper } from '../../tests';

import { FlatList, FlatListProps } from '.';

describe(`FlatList`, () => {
  const props: FlatListProps = {
    testID: 'FlatList',
  };

  const el_container = mockTestID('List', props.testID!);

  it(`render component #${el_container}`, async () => {
    const { findByTestId, toJSON } = render(<FlatList {...props} />, {
      wrapper,
    });
    const sut = await findByTestId(el_container);
    expect(sut).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
