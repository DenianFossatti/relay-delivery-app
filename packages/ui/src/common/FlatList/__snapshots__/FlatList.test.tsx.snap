// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/FlatList render component #List_FlatList 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    accessible={false}
    contentContainerStyle={
      [
        {},
        {
          "dataSet": {},
        },
        {},
      ]
    }
    dataSet={{}}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    style={{}}
    testID="List_FlatList"
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</RNCSafeAreaProvider>
`;
